### ISSUE TITLE
Doublons avec position différente

### ISSUE THREAD

**Issue by cquest** [etalab/ban-data#5]
Exemple: 
rue d'angleterre 0005 12  33327 33720   434177.8 6400245.8
rue d'angleterre 0005 12  33327 33720   434173.9 6400242.3

**Comment by FrancoisVIGN** [comment]
Dans les données de production, il y a bien deux adresses au 12 rue d'Angleterre sur l'INSEE 33327.

**Comment by FrancoisVIGN** [comment]
il faut supprimer le premier

**Comment by cquest** [comment]
Comptez-vous filtrer ceci en amont avant la sortie BAN dans les prochaines livraisons ou doit on le faire en aval ?
Dans le second cas, quelle règle appliquer ? Garder l'adresse avec l'ID le plus élevé (donc récent si j'ai compris la logique d'attribution des ID) ?

**Comment by FrancoisVIGN** [comment]
oui. Je pense que c'est la meilleure solution. Il s'agit de données créées et mises à jour de manières différentes suite à l'intégration de différentes sources qui se sont ensuite rapprochées !

**Comment by ghost** [comment]
On intègre vos commentaires au fur et à mesure

**Comment by RemLagrange** [comment]
J'ai le même souci pour les adresses de Charente-Maritime. Plus de 6000 apparaîssent trois fois ou plus, avec un pic de 68 doublons pour le 1601 ROUTE DES HUTTES 17650 SAINT-DENIS-D'OLÉRON.

Je pense que certains doublons peuvent être expliqués par le non-renseignement des indices de répétition.

**Comment by FrancoisVIGN** [comment]
une autre sources découverte récemment est appartenance à des parcelles cadastrales différentes. si des parcelles sont liées à une même adresse, l'adresse peut apparaître plusieurs fois le long d'une même voie. 
S'il s'agit de problème sur des indices de répétition, il est possible de faire la mise à jour sur le Guichet Adresse Mairie de la BAN (pour les organismes publiques) ou par envoie d'informations complémentaires aux différents partenaires de la BAN.
Dans le cas du 1601, il y a autant de parcelle qu'il y a d'occurrence de l'adresse.

**Comment by cquest** [comment]
Je comprends bien ce qui peut être à l'origine de ces doublons à l'entrée.

Un traitement de dédoublonnage serait fort utile car il n'est pas normal d'avoir ces doublons dans la diffusion publique de la BAN.

Le cas de multiples parcelles liées à une même adresse est un cas très courant qui ne peut pas justifier ces doublons.

Pour les doublons à position identique, je propose qu'on ne sorte que l'ID le plus bas.
Pour les doublons à position différent, il faut regarder d'un peu plus près la variation de la position pour voir si une simple moyenne donne quelque chose de correct.

**Comment by RemLagrange** [comment]
Nous avons poussé nos recherches hier et nous en sommes arrivés à la conclusion de cquest. Même si les adresses redondantes ne sont pas systématiquement identiques au point de vue coordonnées, beaucoup le sont et nécessiteraient un traitement d'épuration.

Malheureusement dans le cas de la  Charente-Maritime, ce phénomène concerne plus de 28000 adresses distinctes, et il est très difficile d'estimer les strictement identiques, de celles ayant simplement des coordonnées différentes.

Pour reprendre mon exemple plus haut, le 1601 Route des Huttes à Saint-Denis-d'Oléron est un parc de stationnement de caravanes, qui possède bien 68 parcelles bien définies, mais dont les coordonnées ne diffèrent pas systématiquement dans le fichier de recensement.

Je retiens la proposition de cquest quant à l'ID à garder en cas de doublons. Je vais tenter de travailler cela de mon côté, mais c'est hélas un travail de longue haleine.
